src/neo/SmartContract/ApplicationEngine.Runtime.cs

```diff
@@ -104,7 +106,19 @@ internal bool CheckWitnessInternal(UInt160 hash)
         {
             if (ScriptContainer is Transaction tx)
             {
-                if (!tx.Cosigners.TryGetValue(hash, out Cosigner cosigner)) return false;
+                IReadOnlyDictionary<UInt160, Cosigner> cosigners;
+                OracleResponse response = tx.Attributes.OfType<OracleResponse>().FirstOrDefault();
+                if (response is null)
+                {
+                    cosigners = tx.Cosigners;
+                }
+                else
+                {
+                    OracleRequest request = NativeContract.Oracle.GetRequest(Snapshot, response.Id);
+                    Transaction request_tx = Snapshot.GetTransaction(request.Txid);
```

Comment:
```suggestion
                    Transaction request_tx = Snapshot.GetTransaction(request.Txid);
                    if (request_tx == null) return false;
```

Reply:
I think it is not necessary. If there is a request, the transaction must be found.